### PR TITLE
initial rubrik module sync

### PR DESCRIPTION
**Pull request by drew-russell**
##### SUMMARY

Initial submission for a series of Rubrik modules. This PR also includes the relevant module_utils and doc_fragements files.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

rubrik_cluster_version

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/module_utils/rubrik_cdm.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    """
+
+    ansible = module.params
+
+    try:
+        node_ip = ansible["node_ip"]
+        username = ansible["username"]
+        password = ansible["password"]
+    except KeyError:
+        node_ip = os.environ.get('rubrik_cdm_node_ip')
```
Comment (gundalow) [inline lib/ansible/module_utils/rubrik_cdm.py]: You can do this all in the argspec
`node_ip=dict(fallback=(env_fallback, ['rubrik_cdm_node_ip'])),
 `

File: lib/ansible/modules/cloud/rubrik/rubrik_cluster_version.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+version_added: '2.8'
+author: Rubrik Build Team (@drew-russell) <build@rubrik.com>
+
+
+extends_documentation_fragment:
+    - rubrik_cdm
+requirements: [rubrik_cdm]
+'''
+
+EXAMPLES = '''
```
Comment (gundalow) [inline lib/ansible/modules/cloud/rubrik/rubrik_cluster_version.py:28]: Please can you include `- name: `'s for each example task

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/rubrik_cdm.py (hunk 1 above)
Comment (drew-russell) [reply lib/ansible/module_utils/rubrik_cdm.py]: TIL. Updated.

File: lib/ansible/modules/cloud/rubrik/rubrik_cluster_version.py (hunk 2 above)
Comment (drew-russell) [reply lib/ansible/modules/cloud/rubrik/rubrik_cluster_version.py:28]: Added `-name` to the example.

**Comment by drew-russell** [comment]
@gundalow looks like the failing test is unrelated to the PR. Can you confirm?

File: lib/ansible/module_utils/rubrik_cdm.py (hunk 1 above)
Comment (gundalow) [reply lib/ansible/module_utils/rubrik_cdm.py]: I need to do a better job at documenting this. Thanks for updating it

**Review by gundalow — CHANGES_REQUESTED** [review]
I didn't notice this before, is there a reason that you support taking credentials at the top-level *and* in `provider`?

**Review by thedoubl3j — COMMENTED** [review]
This PR covers only the modules, spoke with @drew-russell and he will include unit tests for the related in an upcoming PR we can expect either this week or next.

**Comment by drew-russell** [comment]
@gundalow By top level, do yo mean in the task itself? User choice basically. 

There potentially may be a situation where for whatever reason the credentials stored in `provider` may not be appropriate for a specific task and since thats the default option for looking for credentials we wanted the ability for the end user to manually specify node ip, username, and password instead.

**Review by gundalow — APPROVED** [review]

**Comment by thedoubl3j** [comment]
@drew-russell any update on the unit tests?

**Comment by thedoubl3j** [comment]
@drew-russell any update on this? all we are missing are unit tests and a change to version added and we are good to go.

**Comment by drew-russell** [comment]
Closing in favor of collections.

**Pull request closed without merging**